features/netsocket/NetworkStack.h

```diff
@@ -84,10 +100,11 @@ class NetworkStack: public DNS {
      *  The callback should not perform expensive operations such as socket recv/send
      *  calls or blocking operations.
      *
-     *  @param status  NSAPI_ERROR_OK on success, negative error code on failure
+     *  @param status  NSAPI_ERROR_OK on success, negative error code on failure or
+     *                  a positive value  that represents the number of DNS records
```

Comment:
@kjbracey-arm , @SeppoTakalo , @AnttiKauppila , please express your opinion. The callback now acts the following way:
1) If an error occurs - it is returned in `status`
2) If DNS was requested with `gethostbyname`, then `status` contains `NSAPI_ERROR_OK` ( = 0)
3) If DNS was requested with `getaddrinfo` then `status` will contain the number of queries ( = 1, if one address is returned, `NSAPI_ERROR_OK` is not an option in that case).

We have to return the number of addresses somehow, but we don't want to introduce a breaking change. Hence the inconsequent behaviour. Would you recommend a different approach?
@tymoteuszblochmobica , please correct me if I got it wrong.

Reply:
I'm OK with that API.
It is inconsistent, but backward compatible.
User should be aware whether they requested one or multiple responses, and therefore thread the status field accordingly.